org.eclipse.xtext.common.types.ui/src/org/eclipse/xtext/common/types/ui/navigation/LinkToOriginDetector.java

```diff
@@ -84,7 +84,8 @@
 							if (traceToSource == null) {
 								return null;
 							}
-							ILocationInResource sourceInformation = IterableExtensions.head(traceToSource.getAllAssociatedLocations());
+							Iterator<? extends ILocationInResource> allAssociatedLocations = traceToSource.getAllAssociatedLocations().iterator();
```

Comment:
You may want to checkout `com.google.common.collect.Iterables.getFirst(Iterable<? extends T>, T)`